src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -112,10 +117,7 @@ public void onEvent(ResourceOffersEvent event, long sequence,
                     }
                 }
             } else {
-                LOGGER.info("No pending tasks, declining all offers");
-                for (Offer offer : offers) {
-                    driver.declineOffer(offer.getId());
-                }
+                offerLifecycleMgr.addOffers(offers);
             }
```

Comment:
One more thing .. we are going to add unit tests for to cover all these scenarios. So we will uncover bugs.